src/window.rs

```diff
@@ -42,29 +44,36 @@ fn handle_new_grid_size(new_size: LogicalSize, renderer: &Renderer) {
     }
 }
 
-struct WindowWrapper {
-    context: Sdl,
-    window: Window,
-    skulpin_renderer: SkulpinRenderer,
-    renderer: Renderer,
-    mouse_down: bool,
-    mouse_position: LogicalSize,
-    title: String,
-    previous_size: LogicalSize,
-    previous_dpis: (f32, f32),
-    transparency: f32,
-    fullscreen: bool,
+pub struct WindowWrapper {
+    pub context: Sdl,
+    pub window: Option<Window>,
+    pub skulpin_renderer: SkulpinRenderer,
+    pub renderer: Renderer,
+    pub mouse_down: bool,
+    pub mouse_position: LogicalSize,
+    pub title: String,
+    pub previous_size: LogicalSize,
+    pub previous_dpis: (f32, f32),
+    pub transparency: f32,
+    pub fullscreen: bool,
 }
 
 impl WindowWrapper {
-    pub fn new() -> WindowWrapper {
-        let context = sdl2::init().expect("Failed to initialize sdl2");
+    pub fn new(context: Option<Sdl>) -> WindowWrapper {
+        let ctx = match context {
+            Some(ctx) => ctx,
+            None => sdl2::init().expect("Failed to initialize sdl2"),
+        };
+
+        Self::with_context(ctx)
```

Comment:
Minor: I'm not sure why this was extracted into a separate method. Is with_context used outside of the constructor?